### DISCUSSION TITLE
Prevent Sheet close when clicking outside in some conditions

### DISCUSSION THREAD

**Discussion by shrix1** [shadcn-ui/ui#1317] [Q&A]
- Is There any way to prevent Sheet from closing while the user click outside the Sheet component ???
-

**Comment by shrix1** [comment]
Found it we can add open props on Sheet components to fix this Issue

**Reply by Zick-del** [reply]
can you provide an example how it can be done?

**Reply by shrix1** [reply]
<img width="800" alt="Screenshot 2023-12-01 at 1 35 34 AM" src="https://github.com/shadcn-ui/ui/assets/92677078/362baed5-f486-4f33-b1b8-a56a04af4365">
<img width="852" alt="Screenshot 2023-12-01 at 1 36 31 AM" src="https://github.com/shadcn-ui/ui/assets/92677078/18f35274-7c04-45d1-94f2-c08ec3c54044">

See Shadcn is a wrapper around Radix so , this will help you i guess.

**Reply by Zick-del** [reply]
I see.. but i wanted to let the interactions to be able outside of the dialog. For example, the content of the dialog depends on the checkbox that i click outside of the dialog. I guess i'll do it with normal html elements, rather that shadcn component

**Reply by shrix1** [reply]
https://www.radix-ui.com/primitives/docs/components/dialog#api-reference

Refer this doc and play around with it

**Reply by Zick-del** [reply]
Gotcha. Thank you very much!

**Comment by memartello** [comment]
If you want to the sheet be open and keep interaction with you application you can use:

```
 <Sheet modal={false}>
.....
  <SheetContent
                        side={"bottom"}
                        onInteractOutside={(e) => {
                            e.preventDefault();
                        }}>
```

**Reply by joshpachner** [reply]
Thanks @memartello !
Just wanna point out the importance of the modal={false}
I was struggling until I realized that part of your example
